### PR TITLE
Fully support CYP2C19*37 minor alleles

### PR DESCRIPTION
**Pull request by aroon-color**
This PR adds support for `CYP2C19*37` minor alleles. `*37` is defined as partial deletion within CYP2C19 lacking _at least_ `exon 1`. With the expectation that all deletions with `exon 1` result in a frameshift. Below I've explained the rationale for the alleles we're suggesting. 

- `*37.001`-`*37.004` were defined based on **Table 2** in PharmVar's CYP2C19 structural variation document: https://a.storyblok.com/f/70677/x/ae8cc4f43e/cyp2c19_variation_v1-1.pdf

_**Table 2 CYP2C19 Partial Gene Deletions.** All variants lack at least exon 1_

| **Allele** | **Exons deleted** |
|------------|-------------------|
| `*37.001`    |               1-5 |
| `*37.002`    |               1-4 |
| `*37.003`    |                 1 |
| `*37.004`    |               1-7 |

- `*37.005` and `*37.004` are identical but are defined separately due to the `CYP2C19*38` alleles that can result from the deletion. We skip `*37.005` in the Aldy config as a result.
- `*37.006`-`*37.014` are defined based on internal data and variant curation and annotation. While PharmVar defines `*37` alleles as lacking at least exon1, we've expanded this to include other observed deletions that are either frameshifts or remove a domain required for activity. 
  - `*37.009`-`*37.011`, deletions starting at exon2, are out of frame
  - `*37.012`, `del e6`, is out of frame 
  - `*37.013`, `del e6-e7`, is in frame but removes a portion of the heme binding pocket and is likely to result in a no-function allele (PMID 23116231)
  - `*37.014`, `del e8-e9`, is out of frame 
  - `*37.015`, `del exon?`, is a catchall for unknown combinations that are likely to be very rare. All other combinations of exon deletion result in a frameshift

We've verified that these definitions enable calling of all these alleles using historical data.

### CODE REVIEW

File: aldy/resources/genes/cyp2c19.yml [hunk 1]
```diff
@@ -10,597 +10,649 @@ alleles:
...
     activity: no function
     mutations:
-    - [CYP2C19, deletion]
+      - [CYP2C19, deletion]
     ignored: true
   CYP2C19*37.001:
     activity: no function
     mutations:
-    - [CYP2C19, "deletion:e2,i2,e3,i3,e4,i4,e5"]
+      - [CYP2C19, 'deletion:e1,i1,e2,i2,e3,i3,e4,i4,e5']
```
Comment (aroon-color) [inline aldy/resources/genes/cyp2c19.yml:603]: This is the meat of the PR, other changes are all enforcing correct YAML formatting

**Comment by aroon-color** [comment]
@inumanag happy to discuss our methodology for these annotation if you have any questions

**Comment by aroon-color** [comment]
@inumanag Have you had a chance to test these changes?

**Comment by aroon-color** [comment]
Awesome, just pushed a commit which comments out that def

**Comment by inumanag** [comment]
Merged into master manually.

**Pull request closed without merging**